pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -926,3 +940,18 @@ func (r *Builder) GetPopulatorTaskName(pvc *core.PersistentVolumeClaim) (taskNam
 	err = planbase.VolumePopulatorNotSupportedError
 	return
 }
+
+func getDeviceNumber(deviceString string) int {
+	if !strings.HasPrefix(deviceString, "/dev/sd") || len(deviceString) < 8 {
+		// In case we encounter an issue detecting the root disk order,
+		// we will return zero to avoid failing the migration due to boot orde
+		return 0
+	}
+
+	for i := 7; i < len(deviceString); i++ {
```

Comment:
same as above regarding magic numbers:
`for i := len(diskPrefix); i < len(deviceString); i++ {`